tests/acceptance/SendCategoryPurchaseEmailCest.php

```diff
@@ -46,4 +46,31 @@ public function sendCategoryPurchaseEmail(\AcceptanceTester $i) {
     $i->click(Locator::contains('span.subject', $emailSubject));
     $i->waitForText($userEmail, 20);
   }
+
+  public function doNotSendCategoryPurchaseEmail(\AcceptanceTester $i) {
+    $i->wantTo('Buy a product in category and do not receive a "Purchased In This Category" email');
+
+    $productName = 'Category Purchase Test Product';
+    $productFactory = new WooCommerceProduct($i);
+    $categoryId = $productFactory->createCategory('Category 1');
+    $product = $productFactory
+      ->withName($productName)
+      ->withCategoryIds([$categoryId])
+      ->create();
+
+    $emailSubject = 'Product In Category Purchase Test';
+    $newsletterFactory = new Newsletter();
+    $newsletterFactory
+      ->withSubject($emailSubject)
+      ->withAutomaticTypeWooCommerceProductInCategoryPurchased([$product])
+      ->withActiveStatus()
+      ->create();
+
+    $userEmail = Security::generateRandomString() . '-user@email.example';
+    $i->orderProduct($product, $userEmail, true, false);
+
+    $i->amOnMailboxAppPage();
+    $i->dontSee($emailSubject);
```

Comment:
Hmm, this is tricky. It takes some time to deliver the email (see [code for checking it was delivered](https://github.com/mailpoet/mailpoet/blob/master/tests/_support/AcceptanceTester.php#L466)) so if we check immediately after the order was placed the test will pass even though the email was actually sent because we were checking too early.
I guess we will have to wait some time before checking the mailbox, but this will make the test slow.
Maybe we could use [Codeception DB module](https://codeception.com/docs/modules/Db) and create a helper that would check scheduled tasks table and it would wait until there are no scheduled or running sending tasks.
Somethin like `$i->waitUntilAllSendingIsDone()` and we could check the mailbox after that.

Reply:
Hi @costasovo,
I understand, so I added one private function to check if proper email is scheduled for sending in our Help section.
Tested also negative case. So, I hope this method will be fine at least for now, as quick solution to this test.